cmd.c

```diff
@@ -1035,3 +984,54 @@ char *cmd_getnext(char *line, unsigned int *pos, char *blanks, char *cites, char
 
 	return word;
 }
+
+
+int cmd_parseArgs(char *s, char (*args)[LINESZ + 1], u16 *argsc)
+{
+	int i;
+	unsigned int pos = 0;
+
+	for (i = 0; *argsc < MAX_CMD_ARGS_NB; ++i) {
+		cmd_skipblanks(s, &pos, DEFAULT_BLANKS);
+		if (cmd_getnext(s, &pos, DEFAULT_BLANKS, NULL, args[i], sizeof(args[i])) == NULL || *args[i] == 0)
+			break;
+		(*argsc)++;
+	}
+
+	if (!*argsc)
+		return ERR_ARG;
+
+	return 0;
+}
+
+
+int cmd_checkDev(const char *devName, unsigned int *dn)
+{
+	unsigned int i;
+
+	/* Show boot devices if parameter is empty */
+
+	if (*devName == 0) {
+		plostd_printf(ATTR_LOADER, "\nBoot devices: ");
+
+		for (i = 0; cmd_common.devs[i].name; i++)
+			plostd_printf(ATTR_LOADER, "%s ", cmd_common.devs[i].name);
+		plostd_printf(ATTR_LOADER, "\n");
+
+		return ERR_ARG;
+	}
+
+	for (i = 0; cmd_common.devs[i].name; i++)  {
+		if (!plostd_strcmp(devName, cmd_common.devs[i].name))
+			break;
+	}
+
+	if (!cmd_common.devs[i].name) {
```

Comment:
This `for` looks unsafe - are we certain, that we don't go outside of `devs` array? Also, we should compare pointers to NULL.

Reply:
A `devs` array is defined here: https://github.com/phoenix-rtos/plo/blob/ae5fedb90f4d9bc0d6ceffaa6019e29056272e60/armv7a9-zynq7000/low.c#L42-L45 so the last element in the array should be NULL.

This part of the code is derived from : https://github.com/phoenix-rtos/plo/blob/ae5fedb90f4d9bc0d6ceffaa6019e29056272e60/ia32/cmd.c#L168-L174
where objects are not compare to NULL.

In the current code, I have changed to compare pointers to NULL.